GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -1213,46 +1212,36 @@ bool ConfirmEmptyMergeCommit()
-                        Cancelable = true,
+                        AllowCancel = true,
                         Caption = _noFilesStagedCommitCaption.Text,
-                        Icon = TaskDialogStandardIcon.Error,
-                        InstructionText = _noFilesStagedCommitInstructions.Text,
-                        StandardButtons = TaskDialogStandardButtons.Cancel
+                        Icon = TaskDialogIcon.Error,
+                        Heading = _noFilesStagedCommitInstructions.Text,
+                        Buttons = { TaskDialogButton.Cancel }
                     };
 
                     // Option 1: there are no staged files, but there are unstaged files. Most probably user forgot to stage them.
-                    TaskDialogCommandLink lnkStageAndCommit = new("StageAndCommit", Unstaged.IsFilterActive ?
-                        _noFilesStagedCommitAllFilteredUnstagedOption.Text : _noFilesStagedCommitAllUnstagedOption.Text);
+                    string text = Unstaged.IsFilterActive ? _noFilesStagedCommitAllFilteredUnstagedOption.Text : _noFilesStagedCommitAllUnstagedOption.Text;
+                    TaskDialogCommandLinkButton lnkStageAndCommit = new(text, enabled: !Unstaged.IsEmpty)
+                    {
+                        Tag = 1
+                    };
                     lnkStageAndCommit.Click += (s, e) =>
                     {
                         mustStageAll = true;
-                        dialog.Close(TaskDialogResult.Ok);
                     };
-                    dialog.Controls.Add(lnkStageAndCommit);
+                    page.Buttons.Add(lnkStageAndCommit);
 
                     // Option 2: the user just wants to make an empty commmit
-                    TaskDialogCommandLink lnkEmptyCommit = new("MakeEmptyCommit", _noFilesStagedMakeEmptyCommitOption.Text);
-                    lnkEmptyCommit.Click += (s, e) =>
+                    TaskDialogCommandLinkButton lnkEmptyCommit = new(_noFilesStagedMakeEmptyCommitOption.Text)
                     {
-                        dialog.Close(TaskDialogResult.Ok);
-                    };
-                    dialog.Controls.Add(lnkEmptyCommit);
-
-                    dialog.Opened += (s, e) =>
-                    {
-                        /* If there are no unstaged changes, this option must be disabled.
-                         Microsoft.WindowsAPICodePack only allows disabling a CommandLink after the dialog is shown */
-                        if (Unstaged.IsEmpty)
-                        {
-                            lnkStageAndCommit.Enabled = false;
-                        }
+                        Tag = 2
                     };
+                    page.Buttons.Add(lnkEmptyCommit);
 
-                    var dialogResult = dialog.Show();
-                    if (dialogResult == TaskDialogResult.Cancel)
+                    TaskDialogButton result = TaskDialog.ShowDialog(Handle, page);
+                    if (result.Tag is not int)
```

Comment:
`result` is the button instance that was clicked. So this should be something like:
```suggestion
                    if (result != lnkEmptyCommit)
```